File: twidere/src/main/java/org/mariotaku/twidere/util/TwidereViewUtils.java
package org.mariotaku.twidere.util;

import android.support.annotation.UiThread;
import android.view.View;

/**
 * Created by mariotaku on 16/1/23.
 */
public class TwidereViewUtils {
    private TwidereViewUtils() {
    }

    @UiThread
    public static boolean hitView(float x, float y, View view) {
        int[] location = new int[2];
        view.getLocationOnScreen(location);
        return TwidereMathUtils.inRange(x, location[0], location[0] + view.getWidth(), TwidereMathUtils.RANGE_INCLUSIVE_INCLUSIVE) &&
                TwidereMathUtils.inRange(y, location[1], location[1] + view.getHeight(), TwidereMathUtils.RANGE_INCLUSIVE_INCLUSIVE);
    }
}

File: twidere/src/main/kotlin/org/mariotaku/twidere/util/TwidereViewUtils.kt
package org.mariotaku.twidere.util

import android.graphics.RectF
import androidx.annotation.UiThread
import android.view.MotionEvent
import android.view.View

/**
 * Created by mariotaku on 16/1/23.
 */
object TwidereViewUtils {

    private val location = IntArray(2)
    private val rect = RectF()

    @UiThread
    fun hitView(event: MotionEvent, view: View): Boolean {
        view.getLocationOnScreen(location)
        rect.set(location[0].toFloat(), location[1].toFloat(), location[0].toFloat() + view.width,
                location[1].toFloat() + view.height)
        return rect.contains(event.rawX, event.rawY)
    }
}
